include/vast/Translation/CodeGenFallBackVisitor.hpp

```diff
@@ -31,21 +32,54 @@ namespace vast::cg
         }
     };
 
-    template< typename Derived >
+    template< typename Derived
+        , template< typename > typename FallBackMixin = UnsupportedFallBackVisitor
+    >
     struct DefaultFallBackVisitorMixin
         : DefaultFallBackDeclVisitorMixin< Derived >
         , DefaultFallBackStmtVisitorMixin< Derived >
         , DefaultFallBackTypeVisitorMixin< Derived >
-    {
+        , FallBackMixin< Derived >
+	{
         using DeclVisitor = DefaultFallBackDeclVisitorMixin< Derived >;
         using StmtVisitor = DefaultFallBackStmtVisitorMixin< Derived >;
         using TypeVisitor = DefaultFallBackTypeVisitorMixin< Derived >;
 
-        using DeclVisitor::Visit;
-        using StmtVisitor::Visit;
-        using TypeVisitor::Visit;
+        using UnsupportedVisitor = FallBackMixin< Derived >;
+
+        Operation* Visit(const clang::Stmt *stmt) {
+            if (auto result = UnsupportedVisitor::Visit(stmt))
+                return result;
+
+            return StmtVisitor::Visit(stmt);
+        }
+
+        Operation* Visit(const clang::Decl *decl) {
+            if (auto result = UnsupportedVisitor::Visit(decl))
+                return result;
+
+            return DeclVisitor::Visit(decl);
+        }
+
+        Type Visit(const clang::Type *type) {
+            if (auto result = UnsupportedVisitor::Visit(type))
+                return result;
+
+            return TypeVisitor::Visit(type);
+        }
+
+        Type Visit(clang::QualType type) {
+            if (auto result = UnsupportedVisitor::Visit(type))
+                return result;
+
+            return TypeVisitor::Visit(type);
+        }
     };
 
+    template< typename Derived >
+    using DefaultFallbackVisitorConfig = DefaultFallBackVisitorMixin< Derived,
```

Comment:
Keep the default for now as it was, instead provide extra helpers for unsupported one, so you can easily plug it into the whole stack (I think not trying to use it as a default for now will help flesh out the implementation).